node/pkg/reporter/bigtablewriter.go

```diff
@@ -57,9 +57,14 @@ func BigTableWriter(events *AttestationEventReporter, connectionConfig *BigTable
 
 		pubsubClient, err := pubsub.NewClient(ctx, e.connectionConfig.GcpProjectID)
 		if err != nil {
-			return fmt.Errorf("failed to create PubSub client: %w", err)
+			logger.Error("failed to create GCP PubSub client", zap.Error(err))
+			return fmt.Errorf("failed to create GCP PubSub client: %w", err)
```

Comment:
Logging and returning an error in the same place is generally discouraged. [This post](https://www.orsolabs.com/post/go-errors-and-logs/) gives an idea of why at the very top, but just as a general rule, it is not very idiomatic go. One of the golang core developers [has the same opinion](https://dave.cheney.net/tag/error-handling) and has discussed it multiple times.

Log where the error is handled, not inside the function returning the error.

Reply:
Yeah, I agree that is overkill. Wanted to be explicit because it should bubble up to here https://github.com/certusone/wormhole/blob/dev.v2/node/cmd/guardiand/node.go#L726, but after being handled a bit by the Supervisor code.
Wanted to be **VERY** sure that I will be able to find it in logs if it is happening.

Would like to leave it for the short term, then will remove all with the "init success" logs.